src/components/Modals/FilterModal/FilterModal.tsx

```diff
@@ -10,28 +12,27 @@ import './FilterModal.scss';
 
 type TFilterModalProps = {
     isModalOpen: boolean;
-    isToggled: boolean;
     onRequestClose: () => void;
-    onToggle: (value: boolean) => void;
-    selectedPaymentMethods: string[];
-    setSelectedPaymentMethods: (value: string[]) => void;
 };
 
-const FilterModal = ({
-    isModalOpen,
-    isToggled,
-    onRequestClose,
-    onToggle,
-    selectedPaymentMethods,
-    setSelectedPaymentMethods,
-}: TFilterModalProps) => {
+const FilterModal = ({ isModalOpen, onRequestClose }: TFilterModalProps) => {
     const { data } = api.paymentMethods.useGet();
     const { localize } = useTranslations();
+    const { isMobile } = useDevice();
+    const { selectedPaymentMethods, setSelectedPaymentMethods, setShouldUseClientLimits, shouldUseClientLimits } =
+        useStore(
+            useShallow(state => ({
+                selectedPaymentMethods: state.selectedPaymentMethods,
+                setSelectedPaymentMethods: state.setSelectedPaymentMethods,
+                setShouldUseClientLimits: state.setShouldUseClientLimits,
+                shouldUseClientLimits: state.shouldUseClientLimits,
+            }))
+        );
```

Comment:
This is how you can destructure the values from the stores. I preferred to destructure it this way rather than follow the selector function way for each value:

```
const selectedPaymentMethods = useStore((state) => state.selectedPaymentMethods) 
const setSelectedPaymentMethods = useStore((state) => state.setSelectedPaymentMethods)
etc
```